packages/angular/src/generators/setup-mf/lib/setup-host-if-dynamic.ts

```diff
@@ -1,18 +1,40 @@
 import type { Tree } from '@nx/devkit';
-import { joinPathFragments, readProjectConfiguration } from '@nx/devkit';
+import {
+  joinPathFragments,
+  readProjectConfiguration,
+  updateProjectConfiguration,
+} from '@nx/devkit';
 import type { Schema } from '../schema';
 
 export function setupHostIfDynamic(tree: Tree, options: Schema) {
   if (options.federationType === 'static') {
     return;
   }
 
+  const project = readProjectConfiguration(tree, options.appName);
   const pathToMFManifest = joinPathFragments(
-    readProjectConfiguration(tree, options.appName).sourceRoot,
+    project.sourceRoot,
     'assets/module-federation.manifest.json'
   );
 
   if (!tree.exists(pathToMFManifest)) {
     tree.write(pathToMFManifest, '{}');
   }
+
+  const pathToProdWebpackConfig = joinPathFragments(
+    project.root,
+    'webpack.prod.config.js'
+  );
+  if (tree.exists(pathToProdWebpackConfig)) {
+    tree.delete(pathToProdWebpackConfig);
+  }
+
+  project.targets.build.configurations.production = {
+    ...project.targets.build.configurations.production,
+    customWebpackConfig: {
+      path: `${project.root}/webpack.config.js`,
+    },
+  };
```

Comment:
We shouldn't need to repeat the path here, that one is already set in the options, so all configurations will use it unless they overwrite it. We should just delete the overwrite to avoid duplicating the path:
```suggestion
  delete project.targets.build.configurations.production?.customWebpackConfig;
```

Reply:
Alternatively, we could handle this in the `changeBuildTarget` function, so we never set it in the first place if it's unnecessary.

Reply:
good point, i wanted to leave all the logic that handles a dynamic host in the one function, for discoverability. I don't mind though. What do you think is best? If you think it's best to keep it in this place but remove the customWebpackConfig option completely, I'll do that, or if you think it's best to change the changeBuildTarget, I'll do that :)

Reply:
Yeah, that's ok, I don't mind either way. The important bit is to reduce boilerplate when possible.